datafusion/physical-plan/src/aggregates/group_values/group_column.rs

```diff
@@ -224,9 +217,19 @@ where
     where
         B: ByteArrayType,
     {
-        let arr = array.as_bytes::<B>();
-        self.nulls.is_null(lhs_row) == arr.is_null(rhs_row)
-            && self.value(lhs_row) == (arr.value(rhs_row).as_ref() as &[u8])
+        let array = array.as_bytes::<B>();
+
+        // In nullable path, we should check if both `exist row` and `input row`
+        // are null/not null
+        let exist_null = self.nulls.is_null(lhs_row);
+        let input_null = array.is_null(rhs_row);
+        match (exist_null, input_null) {
+            (true, true) => return true,
+            (false, true) | (true, false) => return false,
+            _ => {}
+        }
```

Comment:
how about we extract a common function like `nulls_equal_to`.
And we can also reuse it when impl new `GroupColumn`?

Reply:
good idea -- will do

Reply:
Done in e26a459b3